qr-code/src/main/java/org/odk/collect/qrcode/ScannerOverlay.kt

```diff
@@ -0,0 +1,108 @@
+package org.odk.collect.qrcode
+
+import android.animation.ValueAnimator
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Color
+import android.graphics.Paint
+import android.util.AttributeSet
+import android.view.View
+import kotlin.math.max
+import kotlin.math.roundToInt
+
+class ScannerOverlay(context: Context, attrs: AttributeSet?) :
+    View(context, attrs) {
+
+    private val laserPaint = Paint().also {
+        it.color = Color.RED
+    }
+
+    private val borderPaint = Paint().also {
+        it.color = Color.BLACK
+        it.alpha = 75
+    }
+
+    private val laserAnim = ValueAnimator.ofFloat(0f, 100f).also { animator ->
+        animator.setDuration(1000)
```

Comment:
I wonder if the animation could be faster to match the one from ZXing?

Reply:
Not a big deal to leave it as is but if you play with the color range (see above) you can probably easily address this as well at the same time.